### ISSUE TITLE
Clickhouse: false positives with ALTER TABLE x DROP COLUMN y syntax

### ISSUE THREAD

**Issue by NINNiT** [sqlfluff/sqlfluff#6174]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

My SQL Statement:
`ALTER TABLE x DROP COLUMN y;`

sqlfluff config:
```toml
[sqlfluff]
dialect = clickhouse
templater = jinja
exclude_rules = LT01, CP05, CP03
warnings = RF04

[sqlfluff:indentation]
allow_implicit_indents = True
```
sqlfluff output:
```
== [migrations/shared/V1725613290__change_x_table.sql] FAIL                                                        
L:   1 | P:  30 |  PRS | Line 1, Position 30: Found unparsable section:
                       | 'y'
```

### Expected Behaviour

No failure, as this is valid clickhouse syntax. 
https://clickhouse.com/docs/en/sql-reference/statements/alter/column#drop-column

### Observed Behaviour

False-Positive failures.

### How to reproduce

See the "What happened" section. 

### Dialect

clickhouse

### Version

3.1.1-1

### Configuration

[sqlfluff]
dialect = clickhouse
templater = jinja
exclude_rules = LT01, CP05, CP03
warnings = RF04

[sqlfluff:indentation]
allow_implicit_indents = True

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
